<a href="https://colab.research.google.com/github/vaibhavjiyer87/engine-nvh-deep-learning/blob/main/notebooks/01_dataset_ingestion_and_audit.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
from google.colab import drive
drive.mount("/content/drive")

from pathlib import Path

PROJECT_DRIVE = Path(
    "/content/drive/MyDrive/"
    "NVH_DeepLearning/01_EngineOperatingState"
)

RAW_ROOT = (
    PROJECT_DRIVE
    / "data"
    / "raw"
    / "procedural_engine_sounds"
)

RAW_ROOT.mkdir(parents=True, exist_ok=True)

print(RAW_ROOT)

Mounted at /content/drive
/content/drive/MyDrive/NVH_DeepLearning/01_EngineOperatingState/data/raw/procedural_engine_sounds


In [17]:
# Authenticating Hugging Face HF_TOKEN for the notebook

from google.colab import userdata
from huggingface_hub import login, whoami

hf_token = userdata.get("HF_TOKEN")

if not hf_token:
    raise ValueError(
        "HF_TOKEN was not found. Add it under Colab Secrets "
        "and enable Notebook access."
    )

login(token=hf_token)

account = whoami()
print(f"Authenticated with Hugging Face as: {account['name']}")


Authenticated with Hugging Face as: vaibhavjiyer87


In [27]:
# This download command automatically places the files into the correct
# subfolders

from huggingface_hub import snapshot_download

snapshot_download(
    repo_id="rdoerfler/procedural-engine-sounds",
    repo_type="dataset",
    local_dir=str(RAW_ROOT),
    allow_patterns=[
        "README.md",
        "USAGE.txt",
        "metadata/A_full_set*",
    ],
)

Reconstructing (incomplete total...): |          |  0.00B /  0.00B            

Fetching 2 files:   0%|          | 0/2 [00:00<?, ?it/s]

'/content/drive/MyDrive/NVH_DeepLearning/01_EngineOperatingState/data/raw/procedural_engine_sounds'

In [28]:
# verify the files are downloaded to the right location

for path in sorted(RAW_ROOT.rglob("*")):
    if path.is_file():
        print(path.relative_to(RAW_ROOT))

.cache/huggingface/.gitignore
.cache/huggingface/CACHEDIR.TAG
.cache/huggingface/download/README.md.lock
.cache/huggingface/download/README.md.metadata
.cache/huggingface/download/USAGE.txt.lock
.cache/huggingface/download/USAGE.txt.metadata
.cache/huggingface/trees/923a149597c8943d581f2e50c8473043a551e47a.json
README.md
USAGE.txt
metadata/A_full_set_stats.csv
metadata/A_full_set_summary.json


In [29]:
# Inspect the supplied metadata by loading the statistics csv

import pandas as pd

stats_path = RAW_ROOT / "metadata" / "A_full_set_stats.csv"
stats_df = pd.read_csv(stats_path)

print(stats_df.shape)
display(stats_df.head())
print(stats_df.columns.tolist())

# basic checks
stats_df.info()
display(stats_df.describe(include="all"))

(767, 12)


,filename,samplerate,duration_sec,size_MB,rpm_min,rpm_max,rpm_mean,rpm_std,torque_min,torque_max,torque_mean,torque_std
0,001_Engine-A.wav,48000,12.300000,4.504436,1000.366211,1934.204102,1435.016182,257.553540,-26.519775,-5.279541,-13.552642,5.556298
1,002_Engine-A.wav,48000,12.300000,4.504436,896.911621,1571.350098,1143.547199,222.143582,41.473389,221.984863,165.082997,70.902304
2,003_Engine-A.wav,48000,12.300000,4.504436,3477.783203,6863.403320,5336.456117,1022.023879,-103.027344,165.466309,-22.204808,97.378201
3,004_Engine-A.wav,48000,5.536000,2.027386,876.464844,4018.249512,2127.859435,1050.715227,-69.305420,373.107910,1.924834,88.088522
4,005_Engine-A.wav,48000,11.050667,4.046917,1189.880371,1207.885742,1197.904515,2.736499,157.989502,159.484863,158.658052,0.255127


['filename', 'samplerate', 'duration_sec', 'size_MB', 'rpm_min', 'rpm_max', 'rpm_mean', 'rpm_std', 'torque_min', 'torque_max', 'torque_mean', 'torque_std']
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 767 entries, 0 to 766
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   filename      767 non-null    object 
 1   samplerate    767 non-null    int64  
 2   duration_sec  767 non-null    float64
 3   size_MB       767 non-null    float64
 4   rpm_min       767 non-null    float64
 5   rpm_max       767 non-null    float64
 6   rpm_mean      767 non-null    float64
 7   rpm_std       767 non-null    float64
 8   torque_min    767 non-null    float64
 9   torque_max    767 non-null    float64
 10  torque_mean   767 non-null    float64
 11  torque_std    767 non-null    float64
dtypes: float64(10), int64(1), object(1)
memory usage: 72.0+ KB


,filename,samplerate,duration_sec,size_MB,rpm_min,rpm_max,rpm_mean,rpm_std,torque_min,torque_max,torque_mean,torque_std
count,767,767.0,767.000000,767.000000,767.000000,767.000000,767.000000,767.000000,767.000000,767.000000,767.000000,767.000000
unique,767,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
top,767_Engine-A.wav,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
freq,1,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
mean,NaN,48000.0,11.531774,4.223104,2637.915978,3784.518683,3160.460399,335.673569,51.452637,211.003350,123.233035,41.019819
std,NaN,0.0,1.534565,0.561974,1676.508772,1861.163456,1647.298386,450.490092,164.499231,229.979526,185.354434,74.989892
min,NaN,48000.0,4.117333,1.507854,0.000000,0.000000,0.000000,0.000000,-107.025146,-101.776123,-102.326525,0.000000
25%,NaN,48000.0,11.253333,4.121136,1146.087646,2133.331299,1779.595769,4.580769,-68.450928,-4.806519,-20.753579,1.694241
50%,NaN,48000.0,12.300000,4.504436,2381.896973,3652.648926,2994.674510,183.076537,-13.519287,174.896240,64.231094,4.158461
75%,NaN,48000.0,12.300000,4.504436,3820.190430,5410.919189,4475.540419,475.320529,157.241821,351.425171,223.422025,30.196118


How many files are listed?
*   767

Are all sample rates identical?
* Yes (48000)

What is the duration range?
* 4.1173 sec to 12.3 sec

What RPM range is represented?
* 0 to 7007.446289 rpm

What torque range is represented?
* -107.025146 to 718.475342

Are any metadata fields missing?
* No

Are filenames unique?
* Yes

In [35]:
## Step -5 - Download the first audio subset
# After confirming the metadata looks sensible, download 'A_full_set'

snapshot_download(
    repo_id="rdoerfler/procedural-engine-sounds",
    repo_type="dataset",
    local_dir=str(RAW_ROOT),
    allow_patterns=[
        "README.md",
        "USAGE.txt",
        "metadata/A_full_set/*",
        "dataset/audio/A_full_set/*.wav",
    ],
)

Reconstructing (incomplete total...): |          |  0.00B /  0.00B            

Fetching 769 files:   0%|          | 0/769 [00:00<?, ?it/s]

'/content/drive/MyDrive/NVH_DeepLearning/01_EngineOperatingState/data/raw/procedural_engine_sounds'

In [39]:
# Count the WAV files

DATASET_ROOT = RAW_ROOT / "dataset"
audio_dir = DATASET_ROOT / "audio" / "A_full_set"
wav_files = sorted(audio_dir.glob("*.wav"))

print(f"WAV files found: {len(wav_files)}")
print(wav_files[:3])

WAV files found: 767
[PosixPath('/content/drive/MyDrive/NVH_DeepLearning/01_EngineOperatingState/data/raw/procedural_engine_sounds/dataset/audio/A_full_set/001_Engine-A.wav'), PosixPath('/content/drive/MyDrive/NVH_DeepLearning/01_EngineOperatingState/data/raw/procedural_engine_sounds/dataset/audio/A_full_set/002_Engine-A.wav'), PosixPath('/content/drive/MyDrive/NVH_DeepLearning/01_EngineOperatingState/data/raw/procedural_engine_sounds/dataset/audio/A_full_set/003_Engine-A.wav')]


In [40]:
# Validation cell

from pathlib import Path

DATASET_ROOT = RAW_ROOT / "dataset"
audio_dir = DATASET_ROOT / "audio" / "A_full_set"
wav_files = sorted(audio_dir.glob("*.wav"))

EXPECTED_FILE_COUNT = 767

if not audio_dir.exists():
    raise FileNotFoundError(
        f"Audio directory was not created:\n{audio_dir}\n\n"
        "The download may not have started, the pattern may not have "
        "matched, or RAW_ROOT may point to the wrong location."
    )

if len(wav_files) != EXPECTED_FILE_COUNT:
    raise RuntimeError(
        f"Incomplete download: expected {EXPECTED_FILE_COUNT} WAV files, "
        f"but found {len(wav_files)}."
    )

total_gb = sum(path.stat().st_size for path in wav_files) / (1024**3)

print("A_full_set validation passed.")
print(f"WAV files: {len(wav_files)}")
print(f"Total size: {total_gb:.2f} GiB")
print(f"Location: {audio_dir}")

A_full_set validation passed.
WAV files: 767
Total size: 3.16 GiB
Location: /content/drive/MyDrive/NVH_DeepLearning/01_EngineOperatingState/data/raw/procedural_engine_sounds/dataset/audio/A_full_set
